src/Core/Types.Selection/SelectionVisitorBase.cs

```diff
@@ -111,6 +113,24 @@ protected virtual void LeaveObject(IFieldSelection selection)
         {
         }
 
+        protected IFieldSelection UnwrapPaging(
+            IFieldSelection fieldSelection)
+        {
+            if (fieldSelection.Field.Type.ToClrType() == typeof(IConnection) &&
+                fieldSelection.Field.Type.NamedType() is ObjectType type)
+            {
+                foreach (IFieldSelection selection in Context.CollectFields(
+                    type, fieldSelection.Selection.SelectionSet))
+                {
+                    if (selection.Field.Name == "nodes")
```

Comment:
In the implementation I wrote, I used `nodes` as it was much easier to use in my testing than to use `edges`.  However, from what I am able to determine, the `nodes` node is a convenience node added by HotChocolate to make paging simpler, where implementations using the Relay specification would only know how to deal with choosing whether to use the `nodes` node or the `edges/node` node, or somehow make use of both at the same time, and also document the limitations of how the selection middleware can be used with the Relay paging middleware, and that unsupported applications of the selection middleware on top of the paging middleware can have unexpected behavior.

As it is written now, however, will work just fine for my immediate uses.

Reply:
ah totally forgot about this, I really want refactor this. It should not have a dependency on the field name. I think it was just the easiest way of selecting the subselection in my example

Reply:
That is the same reason why I did the same thing in my implementation.